frontend/src/pages/pipelines/global/experiments/artifacts/ArtifactsTable.tsx

```diff
@@ -0,0 +1,200 @@
+import SimpleDropdownSelect from '~/components/SimpleDropdownSelect';
+import { ArtifactType } from '~/concepts/pipelines/kfTypes';
+import { useMlmdListContext } from '~/concepts/pipelines/context';
+import { FilterOptions, columns, initialFilterData, options } from './constants';
+
+interface ArtifactsTableProps {
+  artifacts: MlmdArtifact.AsObject[] | null | undefined;
+  nextPageToken: string | undefined;
+  isLoaded: boolean;
+}
+
+export const ArtifactsTable: React.FC<ArtifactsTableProps> = ({
+  artifacts,
+  isLoaded,
+  nextPageToken,
+}) => {
+  const {
+    maxResultSize,
+    setFilterQuery,
+    setPageToken: setRequestToken,
+    setMaxResultSize,
+  } = useMlmdListContext(nextPageToken);
+  const [page, setPage] = React.useState(1);
+  const [filterData, setFilterData] = React.useState(initialFilterData);
+  const onClearFilters = React.useCallback(() => setFilterData(initialFilterData), []);
+  const [pageTokens, setPageTokens] = React.useState<Record<number, string>>({});
+
+  const onFilterUpdate = React.useCallback(
+    (key: string, value: string | { label: string; value: string } | undefined) =>
+      setFilterData((prevValues) => ({ ...prevValues, [key]: value })),
+    [],
+  );
+
+  const onNextPageClick = React.useCallback(
+    (_: React.SyntheticEvent<HTMLButtonElement>, nextPage: number) => {
+      if (nextPageToken) {
+        setPageTokens((prevTokens) => ({ ...prevTokens, [nextPage]: nextPageToken }));
+        setRequestToken(nextPageToken);
+        setPage(nextPage);
+      }
+    },
+    [nextPageToken, setRequestToken],
+  );
+
+  const onPrevPageClick = React.useCallback(
+    (_: React.SyntheticEvent<HTMLButtonElement>, prevPage: number) => {
+      if (pageTokens[prevPage]) {
+        setRequestToken(pageTokens[prevPage]);
+        setPage(prevPage);
+      } else {
+        setRequestToken(undefined);
+      }
+    },
+    [pageTokens, setRequestToken],
+  );
+
+  React.useEffect(() => {
+    if (Object.values(filterData).some((filterOption) => !!filterOption)) {
+      let filterQuery = '';
+
+      if (filterData[FilterOptions.Artifact]) {
+        const artifactNameQuery = `custom_properties.display_name.string_value LIKE '%${
+          filterData[FilterOptions.Artifact]
+        }%'`;
+        filterQuery += filterQuery.length ? ` AND ${artifactNameQuery}` : artifactNameQuery;
+      }
+
+      if (filterData[FilterOptions.Id]) {
+        const artifactIdQuery = `id = cast(${filterData[FilterOptions.Id]} as int64)`;
+        filterQuery += filterQuery.length ? ` AND ${artifactIdQuery}` : artifactIdQuery;
+      }
+
+      if (filterData[FilterOptions.Type]) {
+        const artifactTypeQuery = `type LIKE '%${filterData[FilterOptions.Type]}%'`;
+        filterQuery += filterQuery.length ? ` AND ${artifactTypeQuery}` : artifactTypeQuery;
+      }
+
+      setFilterQuery(filterQuery);
+    } else {
+      setFilterQuery('');
+    }
```

Comment:
can this filter logic be abstracted so the executions can reuse this?

Reply:
I can do this in my PR after this is merged.